diff --git a/libdeja/BackendRemote.vala b/libdeja/BackendRemote.vala
index 39bcf524..c84b4fab 100644
--- a/libdeja/BackendRemote.vala
+++ b/libdeja/BackendRemote.vala
@@ -147,6 +147,17 @@ public class BackendRemote : BackendFile
     }
   }
 
+  public void check_fuse(File root) throws Error
+  {
+    if (root.get_path() == null) {
+      throw new IOError.FAILED("%s", _(
+        "You must first install FUSE support for GVfs, " +
+        "likely provided by the gvfs-fuse package. " +
+        "Then log out, log back in, and try again."
+      ));
+    }
+  }
+
   public override async bool mount() throws Error
   {
     if (!Network.get().connected) {
@@ -206,6 +217,7 @@ public class BackendRemote : BackendFile
     try {
       yield root.mount_enclosing_volume(MountMountFlags.NONE, mount_op, null);
     } catch (IOError.ALREADY_MOUNTED e) {
+      check_fuse(root);
       return false;
     } catch (IOError.FAILED_HANDLED e) {
       // needed mount_op but none provided
@@ -221,6 +233,8 @@ public class BackendRemote : BackendFile
         mount_op.@set("retry_mode", false);
     }
 
+    check_fuse(root);
+
     // FIXME: I've found issues with mounting and unmounting often - it seems
     // to confuse FUSE. Which, fair. So just litter the user's session with
     // our mount and return false here.
