diff --git a/data/org.gnome.DejaDup.metainfo.xml.in b/data/org.gnome.DejaDup.metainfo.xml.in
index 5558af12..e41b5b91 100644
--- a/data/org.gnome.DejaDup.metainfo.xml.in
+++ b/data/org.gnome.DejaDup.metainfo.xml.in
@@ -31,7 +31,6 @@ SPDX-FileCopyrightText: Michael Terry
     <kudo>HighContrast</kudo>
     <kudo>ModernToolkit</kudo>
     <kudo>Notifications</kudo>
-    <kudo>UserDocs</kudo>
   </kudos>
 
   <provides>
diff --git a/data/ui/help-overlay.ui b/data/ui/help-overlay.ui
index 0e29fbbf..a8b0f559 100644
--- a/data/ui/help-overlay.ui
+++ b/data/ui/help-overlay.ui
@@ -17,12 +17,6 @@ SPDX-FileCopyrightText: Michael Terry
           <object class="GtkShortcutsGroup">
             <property name="visible">1</property>
             <property name="title" translatable="yes" context="shortcut window">General</property>
-            <child>
-              <object class="GtkShortcutsShortcut">
-                <property name="accelerator">F1</property>
-                <property name="title" translatable="yes" context="shortcut window">Show help</property>
-              </object>
-            </child>
             <child>
               <object class="GtkShortcutsShortcut">
                 <property name="visible">1</property>
diff --git a/data/ui/MainHeaderBar.ui b/data/ui/MainHeaderBar.ui
index 38449842..8af37785 100644
--- a/data/ui/MainHeaderBar.ui
+++ b/data/ui/MainHeaderBar.ui
@@ -16,10 +16,6 @@ SPDX-FileCopyrightText: Michael Terry
         <attribute name="label" translatable="yes">_Keyboard Shortcuts</attribute>
         <attribute name="action">win.show-help-overlay</attribute>
       </item>
-      <item>
-        <attribute name="label" translatable="yes">_Help</attribute>
-        <attribute name="action">app.help</attribute>
-      </item>
       <item>
         <attribute name="label" translatable="yes">_About Backups</attribute>
         <attribute name="action">app.about</attribute>
diff --git a/deja-dup/main.vala b/deja-dup/main.vala
index 54fa8413..bc563442 100644
--- a/deja-dup/main.vala
+++ b/deja-dup/main.vala
@@ -224,7 +224,6 @@ public class DejaDupApp : Gtk.Application
 
   void help()
   {
-    Gtk.show_uri(get_app_window(), "help:" + Config.PACKAGE, Gdk.CURRENT_TIME);
   }
 
   void menu()
